### PR TITLE
vi: Unify resolutions values and accurate implementation of them.

### PR DESCRIPTION
**Pull request by AcK77**
To continue what was made in #2618, I've REd `vi` service a bit. Now values and checks related to displays are more accurate.

- `am`  GetDefaultDisplayResolution / GetDefaultDisplayResolutionChangeEvent have more informations on what the service does.
- `vi:u/vi:m/vi:s` GetDisplayService are now accurate.
- `IApplicationDisplay` GetRelayService, GetSystemDisplayService, GetManagerDisplayService, GetIndirectDisplayTransactionService, ListDisplays, OpenDisplay, OpenDefaultDisplay, CloseDisplay, GetDisplayResolution are now properly implemented.
- Some other calls are cleaned or have extra checks accordingly to RE.

Additionnaly, `IFriendService` have some wrong aligned things, and `pm:info` service placeholder was missing.

### CODE REVIEW

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Comment by Linear524** [comment]
Is it possible to change Native 720p/1080p option to two different options = Native 720p and Native 1080p ?
Currently a lot of games switching renderer to 1080p in docked mode and some of them looks too sharp, before they was fine at 720p (Astral Chain and LM3 for example).
Another thing is - many 15'' inch laptops doesn't need 1080p in emulated games and there is a little fps hit after 1080p PR was merged to Master branch. About 1~2% for average GPU...

**Comment by gdkchan** [comment]
Can't you just disable docked mode for those games?

**Comment by Linear524** [comment]
@gdkchan
Sorry, but if we disable Docked mode - games will jump to power saving mode of real Switch, and it brings not only lowered resolution, but also lowered graphics quality, like low resolution shadows, lower draw distances and lower detail level, etc...
Before 1080p vi update Native mode was at 720p with all graphics games could show and pretty fast on mid-level hardware. And everyone who was interested in resolution scaling was using additional options like 2K, 4K or custom resolution...

**Comment by riperiperi** [comment]
This doesn't make sense, in docked astral chain was targeting 900p, and LM3 was targeting 1080p, and have been since the start. This PR should not change that.

**Comment by Benedani** [comment]
> @gdkchan
> Sorry, but if we disable Docked mode - games will jump to power saving mode of real Switch, and it brings not only lowered resolution, but also lowered graphics quality, like low resolution shadows, lower draw distances and lower detail level, etc...
> Before 1080p vi update Native mode was at 720p with all graphics games could show and pretty fast on mid-level hardware. And everyone who was interested in resolution scaling was using additional options like 2K, 4K or custom resolution...

You can still use resolution scaling to go back to 720p - use a custom scale of 0.75

**Comment by Linear524** [comment]
@riperiperi
Yup, you right ! Those games was always aimed at their intended resolution, but somehow, after PR #2618, Astral Chain shows me 1080p and very crisp flashing edges on characters and other cel-shaded edges... At first I thought that maybe image resizing was broken or bilinear interpolation was changed, or maybe antialiasing function was changed, but then I saw increased resolution... Don't know how to describe it more precise, but something feels different in image output and it now looks a bit different compared what my docked switch console shows on TV... Before that emulated image and original Switch image was the same on my TV... 
Maybe I'm wrong and need to doublecheck my PC settings...

File: Ryujinx.HLE/HOS/Services/Vi/RootService/ApplicationDisplayService/ISystemDisplayService.cs [hunk 1]
```diff
@@ -42,12 +42,17 @@ public ResultCode CreateStrayLayer(ServiceCtx context)
         // GetDisplayMode(u64) -> nn::vi::DisplayModeInfo
         public ResultCode GetDisplayMode(ServiceCtx context)
         {
-            // TODO: De-hardcode resolution.
```
Comment (Thog) [inline Ryujinx.HLE/HOS/Services/Vi/RootService/ApplicationDisplayService/ISystemDisplayService.cs:45]: My first contribution to the project can now retire peacefully 🙏

**Review by Thog — APPROVED** [review]

**Pull request merged**